### PR TITLE
Added supported date format in RuletteInputProcessor and InputDateValue

### PR DESCRIPTION
**Pull request by abhinav3**

### CODE REVIEW

File: rulette-core/src/main/java/com/github/kislayverma/rulette/core/util/RuletteInputProcessor.java [hunk 1]
```diff
@@ -11,6 +13,8 @@
  */
 public class RuletteInputProcessor {
 
+    private static final SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```
Comment (calvincodes) [inline rulette-core/src/main/java/com/github/kislayverma/rulette/core/util/RuletteInputProcessor.java]: This is not ThreadSafe. Use Joda DateTimeFormatter instead.

File: rulette-core/src/main/java/com/github/kislayverma/rulette/core/util/RuletteInputProcessor.java [hunk 2]
```diff
@@ -26,11 +30,17 @@
             if(fieldValue == null){
                 fieldValueString = "";
             }else {
-                fieldValueString = fieldValue.toString();
+                if(fieldValue instanceof Date){
+                    Date date = (Date)fieldValue;
```
Comment (calvincodes) [inline rulette-core/src/main/java/com/github/kislayverma/rulette/core/util/RuletteInputProcessor.java]: Redundant date variable.

File: rulette-mysql-provider/src/main/java/com/github/kislayverma/rulette/mysql/MysqlDataProvider.java [hunk 3]
```diff
@@ -102,6 +108,12 @@ public Rule saveRule(String ruleSystemName, Rule rule) throws SQLException, Exce
             ResultSet resultSet = preparedStatement.executeQuery();
 
             List<Rule> ruleList = convertToRules(resultSet, metaData);
+
+
+            resultSet.close();
```
Comment (calvincodes) [inline rulette-mysql-provider/src/main/java/com/github/kislayverma/rulette/mysql/MysqlDataProvider.java]: @abhinav3: We should do the way it's done in the answer you shared. Try, catch, FINALLY - to ensure we are closing all the resources even if an exception is thrown in between.

File: rulette-core/src/main/java/com/github/kislayverma/rulette/core/ruleinput/value/defaults/InputDateValue.java [hunk 4]
```diff
@@ -11,10 +11,10 @@
     private static final long serialVersionUID = 5666450390675442878L;
 
     private final Date value;
-    private static final DateTimeFormatter formatter = DateTimeFormat.forPattern("yyyyMMdd");
+    private static final DateTimeFormatter formatter = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss");
 
     public InputDateValue (String value) throws Exception {
-        this.value = value == null || value.isEmpty() ? null : formatter.parseDateTime(value).toDate();
+        this.value = (value == null || value.isEmpty()) ? null : formatter.parseDateTime(value.substring(0, 19)).toDate(); //Taking till seconds value of the date string.
```
Comment (kislayverma) [inline rulette-core/src/main/java/com/github/kislayverma/rulette/core/ruleinput/value/defaults/InputDateValue.java]: This should not be in the default implementation. You should implement your own InputDateValue with a specific format and use that at the time of rule system instantiation.

File: rulette-core/src/main/java/com/github/kislayverma/rulette/core/ruleinput/value/defaults/InputDateValue.java [hunk 5]
```diff
@@ -29,7 +29,7 @@ public Date getValue() {
 
     @Override
     public int compareTo(String obj) throws ParseException {
-        return this.value.compareTo(formatter.parseDateTime(obj).toDate());
+        return this.value.compareTo(formatter.parseDateTime(obj.substring(0, 19)).toDate());
```
Comment (kislayverma) [inline rulette-core/src/main/java/com/github/kislayverma/rulette/core/ruleinput/value/defaults/InputDateValue.java]: This also seems specific to your use case

File: rulette-mysql-provider/src/main/java/com/github/kislayverma/rulette/mysql/MysqlDataProvider.java [hunk 6]
```diff
@@ -54,14 +54,20 @@ private Connection getConnection() throws SQLException, IOException {
...
+        Statement statement = connection.createStatement();
         ResultSet resultSet =
             statement.executeQuery("SELECT * " + " FROM " + getRuleSystemMetaData(ruleSystemName).getTableName());
 
         if (resultSet != null) {
             rules = convertToRules(resultSet, getRuleSystemMetaData(ruleSystemName));
         }
 
+
+        resultSet.close();
```
Comment (kislayverma) [inline rulette-mysql-provider/src/main/java/com/github/kislayverma/rulette/mysql/MysqlDataProvider.java]: All these should probably in some finally lock as well

File: rulette-core/src/main/java/com/github/kislayverma/rulette/core/util/RuletteInputProcessor.java [hunk 7]
```diff
@@ -26,11 +33,16 @@
             if(fieldValue == null){
                 fieldValueString = "";
             }else {
-                fieldValueString = fieldValue.toString();
+                if(fieldValue instanceof Date){
+                    fieldValueString = new DateTime(fieldValue).toString(formatter);
+                }else {
+                    fieldValueString = fieldValue.toString();
+                }
             }
```
Comment (kislayverma) [inline rulette-core/src/main/java/com/github/kislayverma/rulette/core/util/RuletteInputProcessor.java]: What does this change do?

Comment (abhinav3) [reply rulette-core/src/main/java/com/github/kislayverma/rulette/core/util/RuletteInputProcessor.java]: It reads Date in the particular format we have specified, as date can be sent in any format in the request payload.

**Pull request merged**